Sources/PrimerSDK/Classes/Data Models/AsyncPaymentMethodTokenizationViewModel.swift

```diff
@@ -0,0 +1,394 @@
+        
+        case .applePay:
+            return nil
+        case .googlePay:
+            return nil
+        case .klarna:
+            return nil
+        case .payPal:
+            return nil
+        case .unknown:
+            return nil
+        }
+    }()
+    
+    lazy var buttonImage: UIImage? = {
+        switch config.type {
+        case .applePay:
+            return UIImage(named: "apple-pay-logo", in: Bundle.primerResources, compatibleWith: nil)?.withRenderingMode(.alwaysTemplate)
+        case .payPal:
+            return UIImage(named: "paypal3", in: Bundle.primerResources, compatibleWith: nil)
+        case .paymentCard:
+            return UIImage(named: "creditCard", in: Bundle.primerResources, compatibleWith: nil)?.withRenderingMode(.alwaysTemplate)
+        case .googlePay:
+            return nil
+        case .goCardlessMandate:
+            return UIImage(named: "rightArrow", in: Bundle.primerResources, compatibleWith: nil)?.withRenderingMode(.alwaysTemplate)
+        case .klarna:
+            return UIImage(named: "klarna-logo", in: Bundle.primerResources, compatibleWith: nil)
+        case .payNLIdeal:
+            return UIImage(named: "iDeal-logo", in: Bundle.primerResources, compatibleWith: nil)?.withRenderingMode(.alwaysTemplate)
+        case .apaya:
+            return UIImage(named: "mobile", in: Bundle.primerResources, compatibleWith: nil)?.withRenderingMode(.alwaysTemplate)
+        case .hoolah:
+            return UIImage(named: "hoolah-logo", in: Bundle.primerResources, compatibleWith: nil)?.withRenderingMode(.alwaysTemplate)
+        case .unknown:
+            return nil
+        }
+    }()
+    
+    lazy var buttonColor: UIColor? = {
+        switch config.type {
+        case .applePay:
+            return .black
+        case .payPal:
+            return UIColor(red: 0.745, green: 0.894, blue: 0.996, alpha: 1)
+        case .paymentCard:
+            return .white
+        case .googlePay:
+            return nil
+        case .goCardlessMandate:
+            return .white
+        case .klarna:
+            return UIColor(red: 1, green: 0.702, blue: 0.78, alpha: 1.0)
+        case .payNLIdeal:
+            return UIColor(red: 204.0/255, green: 0.0, blue: 102.0/255, alpha: 1.0)
+        case .apaya:
+            return .white
+        case .hoolah:
+            return UIColor(red: 214.0/255, green: 55.0/255, blue: 39.0/255, alpha: 1.0)
+        case .unknown:
+            return nil
+        }
+    }()
+    
+    lazy var buttonTitleColor: UIColor? = {
+        switch config.type {
+        case .paymentCard,
+                .goCardlessMandate,
+                .apaya,
+                .unknown:
+            return theme.colorTheme.text1
+        case .applePay,
+                .googlePay,
+                .hoolah,
+                .payNLIdeal,
+                .payPal,
+                .klarna:
+            return nil
+        }
+    }()
+    
+    lazy var buttonBorderWidth: CGFloat = {
```

Comment:
this type of logic could be in its own descriptor class targeting payment method button component.